### ISSUE TITLE
Incorrect `ProfileResult` totalTime?

### ISSUE THREAD

**Issue by hydrogen666** [elastic/elasticsearch#45972]
When using profiling utility of Elasticsearch, i mentioned that sum of `breakdown` with key not end with `_count` not equals to `time_in_nanos` field:
```json
{
  "type": "IndexOrDocValuesQuery",
  "description": "timestamp_long:[1564099200000 TO 9223372036854775807]",
  "time_in_nanos": 4982334261,
  "breakdown": {
    "score": 1853884600,
    "build_scorer_count": 2,
    "match_count": 0,
    "create_weight": 12738,
    "next_doc": 2392986791,
    "match": 0,
    "create_weight_count": 1,
    "next_doc_count": 52205821,
    "score_count": 52205821,
    "build_scorer": 631032068,
    "advance": 6418,
    "advance_count": 1
  }
}
```
In this example sum of each entry with key not end with `_count` is 4877922615, not 4982334261

I read the code about [`ProfilingResult`](https://github.com/elastic/elasticsearch/blob/237650e9c054149fd08213b38a81a3666c1868e5/server/src/main/java/org/elasticsearch/search/profile/ProfileResult.java), `nodeTime` field equals to sum of values of  constructor parameter `timings` map, and `timings` is generated by [`AbstractInternalProfileTree#doGetTree`](https://github.com/elastic/elasticsearch/blob/237650e9c054149fd08213b38a81a3666c1868e5/server/src/main/java/org/elasticsearch/search/profile/AbstractInternalProfileTree.java#L148) method using [`AbstractProfileBreakdown#toTimingMap`](https://github.com/elastic/elasticsearch/blob/7f473b683d4f7426d3aa13cb618b6330a09ba309/server/src/main/java/org/elasticsearch/search/profile/AbstractProfileBreakdown.java#L57):
```java
    public Map<String, Long> toTimingMap() {
        Map<String, Long> map = new HashMap<>();
        for (T timingType : timingTypes) {
            map.put(timingType.toString(), timings[timingType.ordinal()].getApproximateTiming());
            map.put(timingType.toString() + "_count", timings[timingType.ordinal()].getCount());
        }
        return Collections.unmodifiableMap(map);
    }
```
It means that sum of `timing`'s values is not the real time costs total. Shall we estimate entries with key ends with `_count` in `timing` to calculate the REAL time costs total?

**Comment by hydrogen666** [comment]
> Hm, I'm not entirely sure, but I think I see at least one bug: https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/search/profile/ProfileResult.java#L225-L231
> 
> It looks like this wasn't updated when we added method counts, so the counts are being included with the timings as part of the node time. At the very least, that part is incorrect

OK, I'll create a PR for this BUG later.

**Comment by cbuescher** [comment]
I looked into this issue again and see the same behaviour still on 6.8.15, but it seems to be fixed in current 7.12.1.
I believe that #56208 silently fixed the summation issue by moving the nodetime calculation out of `ProfileResult` into `AbstractInternalProfileTree` where AbstractProfileBreakdown#toNodeTime is used, which only sums the timings `getApproximateTiming` results but not the counts. 
A manual check to compare this with a random profile breakdown taken on 6.8.15:

```
{
                "type" : "MatchAllDocsQuery",
                "description" : "*:*",
                "time_in_nanos" : 736842,
                "breakdown" : {
                  "score" : 4167,
                  "build_scorer_count" : 8,
                  "match_count" : 0,
                  "create_weight" : 24771,
                  "next_doc" : 4670,
                  "match" : 0,
                  "create_weight_count" : 1,
                  "next_doc_count" : 8,
                  "score_count" : 4,
                  "build_scorer" : 703213,
                  "advance" : 0,
                  "advance_count" : 0
                }
              }
```
Summing the times _excluding the count_ values: 4167 + 24771 + 4670 + 703213 = 736.821 (wrong)
but incl “_count” values: 736.821 + 8 + 1 + 8 + 4 = 736.842 (equals "time_in_nanos)
So this calculation is still off on 6.8.15

On 7.12 however:
```
{
                "type" : "ConstantScoreQuery",
                "description" : "ConstantScore(DocValuesFieldExistsQuery [field=_primary_term])",
                "time_in_nanos" : 7120818,
                "breakdown" : {
                  "set_min_competitive_score_count" : 0,
                  "match_count" : 0,
                  "shallow_advance_count" : 0,
                  "set_min_competitive_score" : 0,
                  "next_doc" : 7061921,
                  "match" : 0,
                  "next_doc_count" : 60363,
                  "score_count" : 9,
                  "compute_max_score_count" : 0,
                  "compute_max_score" : 0,
                  "advance" : 4833,
                  "advance_count" : 4,
                  "score" : 1435,
                  "build_scorer_count" : 8,
                  "create_weight" : 10040,
                  "shallow_advance" : 0,
                  "create_weight_count" : 1,
                  "build_scorer" : 42589
                }
```
Here the values excluding the "_count" values sum to the correct "time_in_nanos": 7061921 + 4833 + 1435 + 10040 + 42589 = 7.120.818

I think we intend to port #56208 back to 6.8 so I will close this issue but will look into adding a short test (if possible) that checks this calculation going foward.
